config/src/invariant.rs

```diff
@@ -31,3 +37,83 @@ impl Default for InvariantConfig {
         }
     }
 }
+
+impl InlineConfigParser for InvariantConfig {
+    fn config_key() -> String {
+        INLINE_CONFIG_INVARIANT_KEY.into()
+    }
+
+    fn try_merge(&self, configs: &[String]) -> Result<Option<Self>, InlineConfigParserError> {
+        let overrides: Vec<(String, String)> = Self::overrides(configs);
+
+        if overrides.is_empty() {
+            return Ok(None)
+        }
+
+        let mut conf = *self;
```

Comment:
I know this is for the sake of removing duplication but it's not super common to do this in Rust and IMO makes things a bit more disconcerting to read for other people who might not be too familiar to the codebase. Can we use `self` directly?

Reply:
I want to clone`self`. Since `self` is both `Clone + Copy` I just copy by dereferencing `self`.
I can explain in a comment